### PR TITLE
fix(TypeAhead): upgrade typeahead to v3.4.1

### LINKED ISSUE #1366: PF3: react-bootstrap-typeahead version should be bumped to bring in bugfixes
In https://github.com/ManageIQ/manageiq-v2v/pull/876, we are seeing an issue downstream which was reported in https://github.com/ericgio/react-bootstrap-typeahead/issues/381, and fixed in their v3.2.3 release. We should update to `react-bootstrap-typeahead@^3.3.4`, their latest release which still supports bootstrap v3.

### LINKED ISSUE #1491: Typeahead warnings after version update
Getting unexpected console warnings after the library has been updated.
opened an issue to `react-bootstrap-typeahead`: [issue](https://github.com/ericgio/react-bootstrap-typeahead/issues/428)

### PR DESCRIPTION
**Pull request by Laviro**
fix #1491
fix #1366

**What**:

**Additional issues**:

### CODE REVIEW

**Comment by mturley** [comment]
I was in the process of adapting my still-open PR for this dependency https://github.com/patternfly/patternfly-react/pull/1367 to push its version to 3.4.1 when I noticed your PR here. Whoops!

I neglected to update the props in the stories like you did, so I'll go ahead and close mine. You'll need to update some jest snapshots, though.

**Comment by Laviro** [comment]
Oops, feel free to continue in your PR as I am afk,
I will be able to continue only tomorrow and don't want to block you.

**Comment by mturley** [comment]
@Laviro would you mind adding "fix #1366" to your description here since your upgrade solves my issue too?

**Comment by mturley** [comment]
No problem, I'm not blocked by this (my PR was sitting waiting for a reviewer anyway).

File: packages/patternfly-3/patternfly-react/src/components/TypeAheadSelect/TypeAheadSelect.stories.js [hunk 1]
```diff
@@ -82,6 +94,7 @@ TypeAheadSelectStories.addWithInfo('Multiple Selection', () => {
       <br />
       <h2>Multiple Selection</h2>
       <TypeAheadSelect
+        id="some-typeahead"
```
Comment (jeff-phillips-18) [inline packages/patternfly-3/patternfly-react/src/components/TypeAheadSelect/TypeAheadSelect.stories.js]: Are ID's required now?

Comment (Laviro) [reply packages/patternfly-3/patternfly-react/src/components/TypeAheadSelect/TypeAheadSelect.stories.js]: They will be required in the next version,
Currently it just raises an unpleasant deprecation warning each render

**Pull request merged**